src/lib/update-lock.ts

```diff
@@ -75,13 +64,52 @@ const locksTaken: { [lockName: string]: boolean } = {};
 process.on('exit', () => {
 	for (const lockName of _.keys(locksTaken)) {
 		try {
-			lockFile.unlockSync(lockName);
+			lockfile.unlockSync(lockName);
 		} catch (e) {
 			// Ignore unlocking errors
 		}
 	}
 });
 
+/**
+ * Cleans up any Supervisor-created lockfiles, intended to be called
+ * on process startup to remove any lockfiles from previous Supervisor process.
+ *
+ * Supervisor-created lockfiles are distinguished by their UID of 65534,
+ * corresponding to the `nobody` user.
+ */
+export async function cleanupLocks() {
+	// Should be /mnt/root/tmp/balena-supervisor/services
+	const lockfileParentDir = getPathOnHost(BASE_LOCK_DIR);
+
+	try {
+		// Unix `find` command for listing all Supervisor-created lockfiles
+		const findCommand = `find ${lockfileParentDir} -type f -user ${lockfile.NOBODY_UID} -name "*updates.lock"`;
```

Comment:
Here is how I was doing it on my other PR

https://github.com/balena-os/balena-supervisor/blob/7f539547d02515b32cd28d65e542fb06c1e794c6/src/lib/update-lock.ts#L39-L69

Reply:
Changed to performing this `find` in entry.sh